smithy-typescript-codegen/src/main/resources/software/amazon/smithy/typescript/codegen/integration/http-binding-utils.ts

```diff
@@ -1,7 +1,2 @@
 const isSerializableHeaderValue = (value: any): boolean =>
-  value !== undefined &&
-  value !== null &&
-  value !== "" &&
-  (!Object.getOwnPropertyNames(value).includes("length") ||
-    value.length != 0) &&
-  (!Object.getOwnPropertyNames(value).includes("size") || value.size != 0);
+  value != undefined && value != null;
```

Comment:
1. we should keep `!==` as we did before for null and undef checks
2. Are we not worried about `value.size` anymore?

Reply:
1. no
2. yes, value size will be ignored